lib/split/helper.rb

```diff
@@ -44,19 +45,20 @@ def reset!(experiment)
 
     def finish_experiment(experiment, options = {:reset => true})
       return true if experiment.has_winner?
+      experiment.version = Split::Experiment.version_for_experiment_key(ab_user.key_for_experiment(experiment)).to_i
```

Comment:
Is it expected that the version number would be changed here?

Some reasons I ask:

* If so, the line before it is checking if a different version of the experiment has a winner. That will prevent the new version from being prevented.
* Also if so, it's kind of odd that there are different versions of the experiment floating around, and arbitrary points in the code to "fix" it. It would be better if the create of experiment objects was in charge of setting the "correct" version. That way the code for doing so would live in one place.  Is that possible?
* If it's not expected to change, why is it being set here?

Reply:
I think this is in order to ensure that the right version of the experiment, as dictated by whatever is present in the user store, is completed. I will move this line up.